qiskit_quantuminspire/qi_jobs.py

```diff
@@ -45,31 +46,20 @@ def submit(self) -> None:
             self._job_ids.append(str(i))
         self.job_id = "999"  # ID of the submitted batch-job
 
-    def _fetch_job_results(self) -> List[JobResult]:
+    async def _fetch_job_results(self) -> List[JobResult]:
         """Fetch results for job_ids from CJM using api client."""
-        raw_results = []
-        for job_id in self._job_ids:
-            job_result = JobResult(
-                id=int(job_id),
-                metadata_id=1,
-                created_on=datetime(2022, 10, 25, 15, 37, 54, 269823, tzinfo=timezone.utc),
-                execution_time_in_seconds=1.23,
-                shots_requested=100,
-                shots_done=100,
-                results={
-                    "0000000000": 0.270000,
-                    "0000000001": 0.260000,
-                    "0000000010": 0.180000,
-                    "0000000011": 0.290000,
-                },
-                job_id=int(job_id),
-            )
-            raw_results.append(job_result)
-        return raw_results
+        results = None
+        async with ApiClient(config()) as client:
+            results_api = ResultsApi(client)
+            result_tasks = [results_api.read_results_by_job_id_results_job_job_id_get(int(id)) for id in self._job_ids]
+            results = await asyncio.gather(*result_tasks, return_exceptions=True)
+        return results
 
     def result(self) -> Result:
         """Return the results of the job."""
-        raw_results = self._fetch_job_results()
+        if self.status() is not JobStatus.DONE:
```

Comment:
This refers to JobStatus, but it should be referring to BatchJobStatus right?

Reply:
replaced with `self.done()`